dwl.c

```diff
@@ -541,19 +565,37 @@ createnotifyx11(struct wl_listener *listener, void *data)
 	struct wlr_xwayland_surface *xwayland_surface = data;
 	Client *c;
 
-	/* Allocate a Client for this surface */
-	c = xwayland_surface->data = calloc(1, sizeof(*c));
-	c->xwayland_surface = xwayland_surface;
-	c->isx11 = 1;
-	c->bw = borderpx;
+	if (xwayland_surface->override_redirect) {
```

Comment:
There are a bunch of lines here that can be pulled outside the if/else.

Reply:
Refactored.

Reply:
Haha, I was thinking something like 8b7cc8a, but no worries.